src/drivers/intel/cavs/ipc.c

```diff
@@ -219,15 +204,17 @@ void ipc_platform_complete_cmd(void *data)
 	/* unmask Busy interrupt */
 	ipc_write(IPC_DIPCCTL, ipc_read(IPC_DIPCCTL) | IPC_DIPCCTL_IPCTBIE);
 
-#if CONFIG_SUECREEK
+	/* are we about to enter D3 ? */
 	if (ipc->pm_prepare_D3) {
-
+#if CONFIG_SUECREEK
 		//TODO: add support for Icelake
 		while (1)
 			wait_for_interrupt(0);
-	}
-
+#else
```

Comment:
@lyakh how about moving the ower_off to be the first thing in platform_complete_cmd(). That we'd still have the same sequence as before ie power off and then send the host notification?

Reply:
@ranj063 but then what's the point of this patch? :-) The whole idea was to make sure we acknowledge the completion of the IPC to the host, so that it doesn't run onto an IPC timeout.

Reply:
I think this patch is not critical. CTX_SAVE IPC is always executed on core 0 anyway